samples/client/petstore/go/go-petstore/pet_api.go

```diff
@@ -42,8 +42,8 @@ func (a PetApi) AddPet (body Pet) (APIResponse, error) {
   path := a.Configuration.BasePath + "/pet"
 
   // verify the required parameter 'body' is set
-  if &body == nil {
-      return *NewAPIResponseWithError("400 - Bad Request"), errors.New("Missing required parameter 'body' when calling PetApi->AddPet")
+  if reflect.DeepEqual((Pet{}), body) {
```

Comment:
@guohuang Unfortunately, with the general model we're using, this is probably the best we can do. I'm investigating an alternative approach as well, stay tuned.